pkg/predictor_source/inferenceservice_registry.go

```diff
@@ -258,23 +310,62 @@ func (isvcr InferenceServiceRegistry) UpdateStatus(ctx context.Context, predicto
 	inferenceService := &v1beta1.InferenceService{}
 
 	inferenceService.ObjectMeta = predictor.ObjectMeta
-	inferenceService.Status.PredictorStatus = predictor.Status
+
+	inferenceService.Status.ModelStatus.TransitionStatus = v1beta1.TransitionStatus(predictor.Status.TransitionStatus)
+	inferenceService.Status.ModelStatus.ModelRevisionStates = &v1beta1.ModelRevisionStates{
+		ActiveModelState: v1beta1.ModelState(predictor.Status.ActiveModelState),
+		TargetModelState: v1beta1.ModelState(predictor.Status.TargetModelState),
+	}
+	inferenceService.Status.ModelStatus.ModelCopies = &v1beta1.ModelCopies{
+		FailedCopies: predictor.Status.FailedCopies,
+	}
+	if predictor.Status.LastFailureInfo != nil {
+		inferenceService.Status.ModelStatus.LastFailureInfo = &v1beta1.FailureInfo{
+			Location:          predictor.Status.LastFailureInfo.Location,
+			Reason:            v1beta1.FailureReason(predictor.Status.LastFailureInfo.Reason),
+			Message:           predictor.Status.LastFailureInfo.Message,
+			ModelRevisionName: predictor.Status.LastFailureInfo.ModelId,
+			Time:              predictor.Status.LastFailureInfo.Time,
+		}
+	}
 
 	if predictor.Status.Available {
-		inferenceService.Status.Conditions = v1beta1.Conditions{
-			v1beta1.Condition{
-				Type:   "Ready",
-				Status: corev1.ConditionTrue,
-			},
+		url, err := apis.ParseURL(predictor.Status.GrpcEndpoint)
+		if err != nil {
+			return false, err
 		}
-		inferenceService.Status.URL = predictor.Status.GrpcEndpoint
-	} else {
-		inferenceService.Status.Conditions = v1beta1.Conditions{
-			v1beta1.Condition{
-				Type:   "Ready",
-				Status: corev1.ConditionFalse,
-			},
+		inferenceService.Status.URL = url
+
+		if inferenceService.Status.Components == nil {
+			inferenceService.Status.Components = make(map[v1beta1.ComponentType]v1beta1.ComponentStatusSpec)
+		}
+		componentStatus, ok := inferenceService.Status.Components[v1beta1.PredictorComponent]
+		if !ok {
+			componentStatus = v1beta1.ComponentStatusSpec{}
 		}
+		componentStatus.URL = url
+		componentStatus.GrpcURL = url
+		componentStatus.RestURL = predictor.Status.HTTPEndpoint
+		inferenceService.Status.Components[v1beta1.PredictorComponent] = componentStatus
+
+		inferenceService.Status.SetCondition(v1beta1.PredictorReady, &apis.Condition{
+			Type:   v1beta1.PredictorReady,
+			Status: corev1.ConditionTrue,
+		})
+		inferenceService.Status.SetCondition(v1beta1.IngressReady, &apis.Condition{
+			Type:   v1beta1.IngressReady,
+			Status: corev1.ConditionTrue,
+		})
```

Comment:
What's the thinking behind this? We don't currently configure an ingress in the MM case right?

Reply:
Hah, this tripped me up for a while as I was initially trying to set the Ready condition to True directly. In KServe, the Ready condition for an ISvc is dependent on a list of conditions mentioned [here](https://github.com/kserve/kserve/blob/master/pkg/apis/serving/v1beta1/inference_service_status.go#L242) with `IngressReady` being one of those conditions. So, every time I call `InferenceService.Status.SetCondition`, if the prerequisite conditions are True, then `Ready` is set to True. 

But I can probably find an alternative way to set the PredictorReady and Ready conditions without going through the `SetCondition` function. I agree that the IngressReady condition doesn't make much sense in ModelMesh.